### PR TITLE
Add missing type annotations

### LINKED ISSUE #85: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_bno055.py:142
- [ ] adafruit_bno055.py:146
- [ ] adafruit_bno055.py:150
- [ ] adafruit_bno055.py:155
- [ ] adafruit_bno055.py:160
- [ ] adafruit_bno055.py:164
- [ ] adafruit_bno055.py:172
- [ ] adafruit_bno055.py:347
- [ ] adafruit_bno055.py:381
- [ ] adafruit_bno055.py:500
- [ ] adafruit_bno055.py:518
- [ ] adafruit_bno055.py:538
- [ ] adafruit_bno055.py:558
- [ ] adafruit_bno055.py:578
- [ ] adafruit_bno055.py:598
- [ ] adafruit_bno055.py:618
- [ ] adafruit_bno055.py:638
- [ ] adafruit_bno055.py:658
- [ ] adafruit_bno055.py:667
- [ ] adafruit_bno055.py:670
- [ ] adafruit_bno055.py:706
- [ ] adafruit_bno055.py:766
- [ ] adafruit_bno055.py:771
- [ ] adafruit_bno055.py:777
- [ ] adafruit_bno055.py:789
- [ ] adafruit_bno055.py:794
- [ ] adafruit_bno055.py:807
- [ ] adafruit_bno055.py:871
- [ ] adafruit_bno055.py:882
- [ ] adafruit_bno055.py:893
- [ ] adafruit_bno055.py:904
- [ ] adafruit_bno055.py:915

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #85 
submitted for review / comment

### CODE REVIEW

File: adafruit_bno055.py [hunk 1]
```diff
@@ -139,37 +145,41 @@
...
         super().__init__(register_address, struct_format)
         self.scale = scale
 
-    def __get__(self, obj, objtype=None):
+    def __get__(self, obj, objtype=None) -> Tuple[float, float, float]:
         result = super().__get__(obj, objtype)
         return tuple(self.scale * v for v in result)
 
-    def __set__(self, obj, value):
+    def __set__(self, obj, value) -> None:
```
Comment (FoamyGuy) [inline adafruit_bno055.py:156]: should the `obj` and `value` here have types? In the __set__ function a little below here it has `int` and `str` for their types, would it be the same for this one?

File: adafruit_bno055.py [hunk 2]
```diff
@@ -139,37 +145,41 @@
 
 
 class _ScaledReadOnlyStruct(Struct):  # pylint: disable=too-few-public-methods
-    def __init__(self, register_address, struct_format, scale):
+    def __init__(self, register_address: int, struct_format: str, scale: float) -> None:
         super().__init__(register_address, struct_format)
         self.scale = scale
 
-    def __get__(self, obj, objtype=None):
+    def __get__(self, obj, objtype=None) -> Tuple[float, float, float]:
```
Comment (FoamyGuy) [inline adafruit_bno055.py:152]: do `obj` and `objtype` here have types that we could add?

**Review by FoamyGuy — CHANGES_REQUESTED** [review]
Thanks for working on this (and so many others) @tcfranks!

I noted a few arguments without types, but everything else looking good to me beyond those.

File: adafruit_bno055.py (hunk 1 above)
Comment (tcfranks) [reply adafruit_bno055.py:156]: Must of missed them? In ny until the weekend with just my phone, so nothing in this till then

File: adafruit_bno055.py (hunk 2 above)
Comment (tcfranks) [reply adafruit_bno055.py]: this is my "iffy" understanding of objects.....see my changes

**Review by FoamyGuy — APPROVED** [review]
Looks good to me. Thanks again.

**Pull request merged**